src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/ListPackage/ListPackageCommandRunner.cs

```diff
@@ -116,8 +118,8 @@ public async Task ExecuteCommandAsync(ListPackageArgs listPackageArgs)
                                     //Filter the packages for outdated
                                     foreach (var frameworkPackages in packages)
                                     {
-                                        frameworkPackages.TopLevelPackages = frameworkPackages.TopLevelPackages.Where(p => !p.AutoReference && (p.LatestVersion == null || p.ResolvedVersion < p.LatestVersion));
-                                        frameworkPackages.TopLevelPackages = frameworkPackages.TopLevelPackages.Where(p => p.LatestVersion == null || p.ResolvedVersion < p.LatestVersion);
+                                        frameworkPackages.TopLevelPackages = frameworkPackages.TopLevelPackages.Where(p => !p.AutoReference && (p.LatestPackageMetadata == null || p.ResolvedPackageMetadata.Identity.Version < p.LatestPackageMetadata.Identity.Version));
+                                        frameworkPackages.TopLevelPackages = frameworkPackages.TopLevelPackages.Where(p => p.LatestPackageMetadata == null || p.ResolvedPackageMetadata.Identity.Version < p.LatestPackageMetadata.Identity.Version);
```

Comment:
@nkolev92 @zivkan question: lines 121 and 122, why do we filter the `TopLevelPackages` twice with a slightly different yet similar filter? Is this a copy/paste line error, and should the second filter be applied to `TransitivePackages` instead?

See also line 123 below which does an `or` on the same `frameworkPackages.TopLevelPackages.Any()`...
Doesn't sound right to me.

Reply:
🤷‍♂ 

Think it's just a bug.

Reply:
Think this commit fixes it: https://github.com/NuGet/NuGet.Client/pull/2928/commits/28fa13089be92ea3b699d2fc46b4a40ab6c109b8

Reply:
I commented on the new code.